lib/Notes/NoteForm/noteShapes.js

```diff
@@ -10,11 +10,14 @@ export const noteDataShape = PropTypes.shape({
   type: PropTypes.string.isRequired,
 });
 
-export const referredEntityDataShape = PropTypes.shape({
-  id: PropTypes.string.isRequired,
-  name: PropTypes.string.isRequired,
-  type: PropTypes.string.isRequired,
-});
+export const referredEntityDataShape = PropTypes.oneOfType([
+  PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    type: PropTypes.string.isRequired,
+  }),
+  PropTypes.bool
```

Comment:
I'm wondering if `PropTypes.bool` is needed. If yes, then I think it's better to use `PropTypes.bool` in components' prop-types:
```js
referredEntityData: PropTypes.oneOfType([ referredEntityDataShape, PropTypes.bool ])
```

Reply:
agree, good point